lib/kdl/include/kdl/set_adapter.h

```diff
@@ -0,0 +1,925 @@
+        using const_reference = const value_type&;
+        using pointer = typename std::allocator_traits<allocator_type>::pointer;
+        using const_pointer = typename std::allocator_traits<allocator_type>::const_pointer;
+        using iterator = typename std::remove_reference<C>::type::iterator;
+        using const_iterator = typename std::remove_reference<C>::type::const_iterator;
+        using reverse_iterator = typename std::remove_reference<C>::type::reverse_iterator;
+        using const_reverse_iterator = typename std::remove_reference<C>::type::const_reverse_iterator;
+    public:
+        using base::const_set_adapter;
+
+        using base::get_allocator;
+
+        using base::begin;
+        using base::cbegin;
+
+        using base::end;
+        using base::cend;
+
+        using base::rbegin;
+        using base::crbegin;
+
+        using base::rend;
+        using base::crend;
+
+        using base::empty;
+        using base::size;
+        using base::max_size;
+
+        using base::count;
+
+        using base::find;
+        using base::equal_range;
+        using base::lower_bound;
+        using base::upper_bound;
+
+        using base::key_comp;
+        using base::value_comp;
+
+        using base::capacity;
+        using base::get_data;
+    public:
+        /**
+         * Assigns the values in the given initializer list to this set. The set is cleared and the given values are
+         * inserted.
+         *
+         * @param values the values to insert
+         * @return a reference to this set
+         */
+        set_adapter& operator=(std::initializer_list<value_type> values) {
+            m_data = values;
+            detail::sort_unique(m_data, m_cmp);
+            return *this;
+        }
+
+        /**
+         * Returns an iterator to the first element in the set. If the set is empty, the returned iterator will be equal to
+         * end().
+         */
+        iterator begin() noexcept {
+            return std::begin(m_data);
+        }
+
+        /**
+         * Returns an iterator to the element following the last element in the set (past-the-end iterator).
+         */
+        iterator end() noexcept {
+            return std::end(m_data);
+        }
+
+        /**
+         * Returns a reverse iterator to the first element of the reversed set. If the container is empty, the returned
+         * iterator will be equal to rend().
+         */
+        reverse_iterator rbegin() noexcept {
+            return std::rbegin(m_data);
+        }
+
+        /**
+         * Returns a reverse iterator to the element following the last element of the reversed container.
+         */
+        reverse_iterator rend() noexcept {
+            return std::rend(m_data);
+        }
+
+        /**
+         * Clears all values from this set.
+         *
+         * Postcondition: empty()
+         */
+        void clear() {
+            m_data.clear();
+            assert(check_invariant());
```

Comment:
check_invariant() should be behind a #ifdef SET_ADAPTER_DEBUG, IMO. 
Having all of the methods O(n) times slower in debug builds is annoying, especially if we're going to be using this a lot.

Reply:
done